src/goto-instrument/code_contracts.cpp

```diff
@@ -129,11 +130,18 @@ void code_contractst::check_apply_invariants(
       t->location_number > loop_end->location_number)
       loop_end = t;
 
-  // see whether we have an invariant
+  // see whether we have an invariant and a variant
   exprt invariant = static_cast<const exprt &>(
     loop_end->get_condition().find(ID_C_spec_loop_invariant));
-  if(invariant.is_nil())
-    return;
+  exprt variant = static_cast<const exprt &>(
+    loop_end->get_condition().find(ID_C_spec_decreases));
+  if(invariant.is_nil()) 
+  {
+    if (variant.is_nil())
+      return;
+    else 
```

Comment:
I'd also log a warning here that we are using a default `true` invariant . It's not unsound -- if insufficient this would fail the checks; but just might be surprising to users.

Reply:
Yes, I will do it.